src/analysisd/rules.c

```diff
@@ -2076,7 +2076,7 @@ STATIC char *loadmemory(char *at, const char *str, OSList* log_msg)
             return (NULL);
         }
 
-        strncat(at, str, strsize);
+        strncat(at, str, strsize + 1);
```

Comment:
How does this work? What does the `+ 1` do here?

Reply:
Maybe we could add an explanation to the PR for future reference :)

Reply:
Hi @RamiroRD

This implementation can produce overflow if the allocation size above is changed, so I have changed it.

Best regards
Hanes